File: Daily Challanges/2026-06-June/2026-06-20.js
function primeFactorization(n) {
  const factors = [];
  let d = 2;

  while (d * d <= n) {
    while (n % d === 0) {
      factors.push(d);
      n = Math.floor(n / d);
    }
    d++;
  }

  if (n > 1) {
    factors.push(n);
  }

  return factors;
}

console.log(primeFactorization(20));
console.log(primeFactorization(17));
console.log(primeFactorization(15));
console.log(primeFactorization(35));
console.log(primeFactorization(999));
console.log(primeFactorization(360));
console.log(primeFactorization(510510));

File: Daily Challanges/2026-06-June/2026-06-20.py
def prime_factorization(n):
    factors = []
    d = 2

    while d * d <= n:
        while n % d == 0:
            factors.append(d)
            n //= d
        d += 1

    if n > 1:
        factors.append(n)

    return factors


print(prime_factorization(20))
print(prime_factorization(17))
print(prime_factorization(15))
print(prime_factorization(35))
print(prime_factorization(999))
print(prime_factorization(360))
print(prime_factorization(510510))
